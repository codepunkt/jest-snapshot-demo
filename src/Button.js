import React from 'react'

export default ({ pressed, text }) => {
  const className = pressed
    ? 'button--pressed'
    : 'button'

  return (
    <button className={className}>
      {text}
    </button>
  )
}
