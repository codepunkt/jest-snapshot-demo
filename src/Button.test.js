import React from 'react'
import { shallow } from 'enzyme'
import Button from './Button'

describe('<Button />', () => {
  it('matches previous snapshot', () => {
    const ary = [
      <Button pressed={true} />,
      <Button pressed={false} />,
      <Button text="Press me!" />,
    ]

    ary.forEach(elem =>
      expect(shallow(elem)).toMatchSnapshot()
    )
  })
})
