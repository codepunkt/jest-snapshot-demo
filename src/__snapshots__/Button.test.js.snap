// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches previous snapshot 1`] = `
<button
  className="button--pressed"
>
  default
</button>
`;

exports[`<Button /> matches previous snapshot 2`] = `
<button
  className="button"
>
  default
</button>
`;

exports[`<Button /> matches previous snapshot 3`] = `
<button
  className="button"
>
  Press me!
</button>
`;
